src/mlpack/methods/linear_regression/linear_regression.cpp

```diff
@@ -95,7 +100,7 @@ void LinearRegression::Predict(const arma::mat& points,
   {
     // We want to be sure we have the correct number of dimensions in the
     // dataset.
-    Log::Assert(points.n_rows == parameters.n_rows - 1);
+    util::CheckSameDimensionality(points, (size_t) (parameters.n_rows - 1), "LinearRegression::Predict()", "points");
```

Comment:
Please make sure you don't run out of the 80 character line limit; see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#line-length-and-wrapping for more details.